langchain/utilities/powerbi.py

```diff
@@ -1,25 +1,22 @@
 """Wrapper around a Power BI endpoint."""
-
 from __future__ import annotations
 
 import logging
 import os
 from copy import deepcopy
-from typing import TYPE_CHECKING, Any, Dict, Iterable, List, Optional, Union
+from typing import Any, Dict, Iterable, List, Optional, Union
 
 import aiohttp
 import requests
 from aiohttp import ServerTimeoutError
+from azure.core.credentials import TokenCredential
```

Comment:
This change makes it such that the code breaks on import for all users that don't have `azure` installed (and `azure` will not be installed for most users as it is an optional dependency).

I outlined changes to the poetry file below that will help specify the optionality of the dependency correctly